case-server/src/main/java/com/powsybl/caseserver/CaseService.java

```diff
@@ -223,6 +226,31 @@ UUID importCase(MultipartFile mpf, boolean toPublic) {
         return caseUuid;
     }
 
+    UUID createCase(UUID sourceCaseUuid) {
+        try {
+            Path existingCaseFile = getCaseFile(sourceCaseUuid);
+            if (existingCaseFile == null || existingCaseFile.getParent() == null) {
+                throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Source case " + sourceCaseUuid + " not found");
+            }
+
+            UUID newCaseUuid = UUID.randomUUID();
+            Path newCaseUuidDirectory = existingCaseFile.getParent().getParent().resolve(newCaseUuid.toString());
```

Comment:
why do we get parent 2 times ?

Reply:
Cases files are stored according to the following folder hierarchy :

private or public (folder) > {caseUuid} (folder) > {caseFile} (file)

When we call getParent from an existing case file it returns its direct parent which is a folder named after the case UUID.
So in order to create a new case we have to call getParent two times to create its folder from the private or public folder.